dimscord/objects/macros.nim

```diff
@@ -60,3 +64,49 @@ macro optionIf*(check: typed): untyped =
 
     result = quote do:
         if `check`: none `varType` else: some (`variable`)
+
+macro construct*(data: JsonNode, kind: typedesc, args: static[openarray[string]]): untyped =

```

Comment:
See above ^. We talked about this in DMs, I did say I would have the macros in the next release.

Reply:
thought it would be best to be put in early so that it can be used more

Reply:
Hm, true. But the thing is, it is an internal breaking change. I want to make sure code stays consistent (For better quality code), so it is better understandable for anyone seeing this repository to also reference from it.

Reply:
Not exactly internal breakage since the old code can stay the same and the new procs follow the same style signature which was the main reason for creating the macro over using a different json system.
But I can remove it and use normal object constructions like the rest of the library if you want?

Reply:
Well, not yet either in the next release you can. If you do leave it, I don't want the code to exactly look again inconsistent. Great work with the macro though.